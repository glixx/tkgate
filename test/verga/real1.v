module top;
  real n,a,b,c;
  integer i;
  reg [31:0] r;

  initial
    begin
	n = 8'hf;
	$display("The number is %g (should be 15)",n);
	n = 1.1 + 4;
	$display("The number is %g (should be 5.1)",n);


	a = 1.2;
	b = 2.3;
	c = a + b;
	$display("The number is %g (should be 3.5)",c);

	$display("The number is %g (should be 12.65)",(1.2+1.1)*(2.3+3.2));

	a = 1;	
	b = 3;
	$display("The number is %g (should be 0.333333)",a/b);
	$display("The number is %g (should be -1)",-a);
	$display("The number is %g (should be 2)",b-a);

	a = 1.2;
	i = 1;
	$display("Expecting 1, got ",i);
	$display("Expecting 1.2, got ",a);
	r = 1.2;
	$display("Expecting 32'h1, got ",r);
	a = r;
	$display("Expecting 1, got ",a);
    end
endmodule
