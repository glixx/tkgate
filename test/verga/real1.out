The number is 15 (should be 15)
The number is 5.1 (should be 5.1)
The number is 3.5 (should be 3.5)
The number is 12.65 (should be 12.65)
The number is 0.333333 (should be 0.333333)
The number is -1 (should be -1)
The number is 2 (should be 2)
Expecting 1, got 1
Expecting 1.2, got 1.2
Expecting 32'h1, got 32'h1
Expecting 1, got 1
